### PR TITLE
ML-149: IJ inspector based on code server

### PR DESCRIPTION
**Pull request by tiginamaria**
[ML-149](https://vyahhi.myjetbrains.com/youtrack/issue/ML-149) Hyperstyle / Server for code inspection based on PyCharm

On this PR we try to support IJ inspections. We implemented a client for the CodeServer - a light IJ running on the server which supports a part of IJ features (e.x. IJ code inspections).

### CODE REVIEW

File: hyperstyle/src/python/review/inspectors/ij/ij_inspections.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def __init__(self, language: Language):
+        self.host = os.environ.get(CODE_SERVER_HOST, "0.0.0.0")
+        self.port = os.environ.get(CODE_SERVER_PORT, 8080)
+        self.root = os.environ.get(CODE_SERVER_PORT, "code/server/api/v1/")
+        self.languageId = LANGUAGE_TO_ID[language]
+
+    def inspect(self, path: Path, config: Dict[str, Any]) -> List[BaseIssue]:
+        with open(path, "r") as code_file:
+            code = code_file.read()
```
Comment (nbirillo) [inline hyperstyle/src/python/review/inspectors/ij/ij_inspections.py:36]: We already have a function for it - https://github.com/hyperskill/hyperstyle/blob/main/hyperstyle/src/python/review/common/file_system.py#L99

File: hyperstyle/src/python/review/inspectors/ij/model.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    line_number: int
+    offset: int
+    length: int
+
+    def to_base_issue(self) -> BaseIssue:
+        return BaseIssue(
+            origin_class='',
+            type=IssueType.CODE_STYLE,  # TODO: make mapping
+            description=self.name,
+            file_path=Path(),  # TODO: make optional?
```
Comment (nbirillo) [inline hyperstyle/src/python/review/inspectors/ij/model.py:31]: Probably you have this path if you call the inspect method with a file

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great! Also, please redirect the PR into the develop branch

**Review by nbirillo — APPROVED** [review]

File: hyperstyle/src/python/review/inspectors/ij/ij_inspections.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+    def inspect_in_memory(self, code: str, config: Dict[str, Any]) -> List[BaseIssue]:
+        return self._get_inspection_result(code).to_base_issues(Path(""))
+
+    def _get_inspection_result(self, code: str) -> IJInspectionResult:
+        try:
+            response = requests.get(
+                f"http://{self.host}:{self.port}/{self.root}inspect",
+                headers={"Content-Type": "application/json"},
+                data=IJCode(code, self.languageId).to_json(),
```
Comment (andrvb) [inline hyperstyle/src/python/review/inspectors/ij/ij_inspections.py:47]: Requests to the external service is better to do with a timeout.

**Pull request merged**